docs/docs/introduction/getting-started.mdx

```diff
@@ -37,91 +37,121 @@ bower install --save recoil
+const App = () => {
 
-```jsx
-function CharacterCounter() {
   return (
-    <div>
-      <TextInput />
-      <CharacterCount />
-    </div>
+         <main>
+              <TextInput /> 
+              <CharCout />
+         </main>
+   
   );
 }
 
-function TextInput() {
-  const [text, setText] = useRecoilState(textState);
+export default App;
 
-  const onChange = (event) => {
-    setText(event.target.value);
-  };
+We will then naviagate to our src folder to create Those components
+We'll implement the `CharacterCounter` component in the following section.
 
-  return (
-    <div>
-      <input type="text" value={text} onChange={onChange} />
-      <br />
-      Echo: {text}
-    </div>
-  );
-}
+## Atom
+
+An **atom** represents a piece of **state**. Atoms can be read from and written to from any component. Components that read the value of an atom are implicitly **subscribed** to that atom, so any atom updates will result in a re-render of all components subscribed to that atom:
+
+We would start here with the TextInput component
+
+```javascript
+    import React from 'react'
+    import {atom, useRecoilState} from 'recoil';
+
+    export const inputState = atom({
+        key: "inputState",
+        default: "",
+      });
+
+
+    function TextInput() {
+        const [input, setInput] = useRecoilState(inputState);
+
+        const onChange = (event) => setInput(event.target.value)
+
+      return (
+        <div>
+            <h1>Text Input</h1>
+                <input
+                    type='text' 
+                    value={input}
+                    onChange={onChange}
+                />
+                <br />
+                {'Echo'}
+                <br />
+                {input}
+        </div>
+      )
+    }
+
+
+    export default TextInput
 ```
 
 ## Selector
 
 A **selector** represents a piece of **derived state**. Derived state is a **transformation** of state. You can think of derived state as the output of passing state to a pure function that modifies the given state in some way:
 
+We would then proceed to our CharCount component 
 ```jsx
-const charCountState = selector({
-  key: 'charCountState', // unique ID (with respect to other atoms/selectors)
-  get: ({get}) => {
-    const text = get(textState);
 
-    return text.length;
-  },
-});
-```
+import { selector, useRecoilValue } from 'recoil';
+import { inputState } from './TextInput';
 
-We can use the `useRecoilValue()` hook to read the value of `charCountState`:
 
-```jsx
-function CharacterCount() {
-  const count = useRecoilValue(charCountState);
+const charCounterText = selector({
+    key: 'charCounterText',
+    get: ({get}) => {
+        const text = get(inputState)
+        return text.length;
+    },
+    
+});
 
-  return <>Character Count: {count}</>;
+we can use the `useRecoilValue()` hook to read the value of `charCounterText`
+const CharCout = () => {
```

Comment:
For documentation prefer `function CharacterCount() {` or `export default function CharacterCount() {`